syntax = "proto3";
option cc_enable_arenas = true;
package NetProto;

import "Common.proto";

message RspFreeHero
{
	uint64 red_hero_id = 1;
	uint64 blue_hero_id = 2;
}

message SelectHeroReq
{
	uint64 hero_id = 1;
}

message SelectHeroRsp
{
	uint64 hero_id = 1;
	bool is_succ = 2;
}

enum SceneObjectType
{
	Building = 0;
	Hero = 1;
	Npc = 2;
	
	MAX = 3;
}

message SceneObjectState
{
	uint64 objid = 1;
	SceneObjectType obj_type = 2;
	int32 model_id = 3;
	PBVector3 pos = 4;
}

message AllSceneObjectState
{
	repeated SceneObjectState objs = 1;
}

message MoveObjectState
{
	int32 radius = 1;
	int32 height = 2;
	int32 mass = 3;
	int32 max_speed = 4;
}

message MoveObjectMutableState
{

}

message MoveToPos
{
	PBVector2 pos = 1;
}
